superset/cli/importexport.py

```diff
@@ -60,338 +58,133 @@ def import_directory(directory: str, overwrite: bool, force: bool) -> None:
-        "--print_stdout",
-        "-p",
-        is_flag=True,
-        default=False,
-        help="Print YAML to stdout",
-    )
-    @click.option(
-        "--back-references",
-        "-b",
-        is_flag=True,
-        default=False,
-        help="Include parent back references",
-    )
-    @click.option(
-        "--include-defaults",
-        "-d",
-        is_flag=True,
-        default=False,
-        help="Include fields containing defaults",
-    )
-    def export_datasources(
-        datasource_file: Optional[str],
-        print_stdout: bool = False,
-        back_references: bool = False,
-        include_defaults: bool = False,
-    ) -> None:
-        """Export datasources to YAML"""
-        # pylint: disable=import-outside-toplevel
-        from superset.utils import dict_import_export
 
-        data = dict_import_export.export_to_dict(
-            session=db.session,
-            recursive=True,
-            back_references=back_references,
-            include_defaults=include_defaults,
+@click.command()
+@with_appcontext
+@click.option(
+    "--datasource-file",
+    "-f",
+    help="Specify the file to export to",
+)
+def export_datasources(datasource_file: Optional[str] = None) -> None:
+    """Export datasources to ZIP file"""
+    # pylint: disable=import-outside-toplevel
+    from superset.commands.dataset.export import ExportDatasetsCommand
+    from superset.connectors.sqla.models import SqlaTable
+
+    g.user = security_manager.find_user(username="admin")
+
+    dataset_ids = [id_ for (id_,) in db.session.query(SqlaTable.id).all()]
+    timestamp = datetime.now().strftime("%Y%m%dT%H%M%S")
+    root = f"dataset_export_{timestamp}"
+    datasource_file = datasource_file or f"{root}.zip"
+
+    try:
+        with ZipFile(datasource_file, "w") as bundle:
+            for file_name, file_content in ExportDatasetsCommand(dataset_ids).run():
+                with bundle.open(f"{root}/{file_name}", "w") as fp:
+                    fp.write(file_content.encode())
+    except Exception:  # pylint: disable=broad-except
+        logger.exception(
+            "There was an error when exporting the datasets, please check "
+            "the exception traceback in the log"
         )
-        if print_stdout or not datasource_file:
-            yaml.safe_dump(data, sys.stdout, default_flow_style=False)
-        if datasource_file:
-            logger.info("Exporting datasources to %s", datasource_file)
-            with open(datasource_file, "w") as data_stream:
-                yaml.safe_dump(data, data_stream, default_flow_style=False)
-
-    @click.command()
-    @with_appcontext
-    @click.option(
-        "--path",
-        "-p",
-        help="Path to a single JSON file or path containing multiple JSON "
-        "files to import (*.json)",
-    )
-    @click.option(
-        "--recursive",
-        "-r",
-        is_flag=True,
-        default=False,
-        help="recursively search the path for json files",
```

Comment:
Were some options removed? I think we may still need the import from json to allow people to migrate from an older version still?